tests/php/test-amp-tumblr-embed-handler.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * Class AMP_Tumblr_Embed_Handler_Test
+ *
+ * @package AMP
+ */
+
+/**
+ * Tests for Tumblr embeds.
+ *
+ * @coversDefaultClass \AMP_Tumblr_Embed_Handler
+ */
+class AMP_Tumblr_Embed_Handler_Test extends WP_UnitTestCase {
+
+	/**
+	 * Set up.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		// Mock the HTTP request.
+		add_filter( 'pre_http_request', [ $this, 'mock_http_request' ], 10, 3 );
+	}
+
+	/**
+	 * Tear down.
+	 */
+	public function tearDown() {
+		remove_filter( 'pre_http_request', [ $this, 'mock_http_request' ] );
+		parent::tearDown();
+	}
+
+	/**
+	 * Mock HTTP request.
+	 *
+	 * @param mixed  $pre Whether to preempt an HTTP request's return value. Default false.
+	 * @param mixed  $r   HTTP request arguments.
+	 * @param string $url The request URL.
+	 * @return array Response data.
+	 */
+	public function mock_http_request( $pre, $r, $url ) {
+		if ( in_array( 'external-http', $_SERVER['argv'], true ) ) {
+			return $pre;
+		}
+
+		if ( false === strpos( $url, 'tumblr.com' ) ) {
+			return $pre;
+		}
+
+		$body = '{"cache_age":3600,"url":"https:\/\/ifpaintingscouldtext.tumblr.com\/post\/92003045635\/grant-wood-american-gothic-1930","provider_url":"https:\/\/www.tumblr.com","provider_name":"Tumblr","author_name":"If Paintings Could Text","version":"1.0","author_url":"https:\/\/ifpaintingscouldtext.tumblr.com\/","type":"rich","html":"\u003Cdiv class=\u0022tumblr-post\u0022 data-href=\u0022https:\/\/embed.tumblr.com\/embed\/post\/2JT2XTaiTxO08wh21dqQrw\/92003045635\u0022 data-did=\u00227ce4825965cbd8bfd208f6aae43de7a528859aee\u0022  \u003E\u003Ca href=\u0022https:\/\/ifpaintingscouldtext.tumblr.com\/post\/92003045635\/grant-wood-american-gothic-1930\u0022\u003Ehttps:\/\/ifpaintingscouldtext.tumblr.com\/post\/92003045635\/grant-wood-american-gothic-1930\u003C\/a\u003E\u003C\/div\u003E\u003Cscript async src=\u0022https:\/\/assets.tumblr.com\/post.js\u0022\u003E\u003C\/script\u003E","height":null,"width":540}';
+
+		return [
+			'body'     => $body,
+			'response' => [
+				'code'    => 200,
+				'message' => 'OK',
+			],
+		];
+	}
+
+	/**
+	 * Get conversion data.
+	 *
+	 * @return array
+	 */
+	public function get_conversion_data() {
+		return [
+			'no_embed'   => [
+				'<p>Hello world.</p>',
+				'<p>Hello world.</p>' . PHP_EOL,
+			],
+
+			'url_simple' => [
+				'https://ifpaintingscouldtext.tumblr.com/post/92003045635/grant-wood-american-gothic-1930' . PHP_EOL,
```

Comment:
For some reason I'm seeing not seeing `post.js` script being reported as a validation error not for this embed. I do see it being reported for another, however: https://teded.tumblr.com/post/184736320764/how-do-vaccines-work

Reply:
Oh yes, it's not being reported for that one. I forgot to add https://teded.tumblr.com/post/184736320764/how-do-vaccines-work to the test data.

Reply:
But why wouldn't it be reported for that one? That's strange because I see it is in the non-AMP version for each:

```html
<figure class="wp-block-embed is-type-rich is-provider-tumblr wp-block-embed-tumblr">
	<div class="wp-block-embed__wrapper">
		<div class="tumblr-post" data-href="https://embed.tumblr.com/embed/post/O6_eRR6K-z9QGTzdU5HrhQ/184736320764" data-did="523d09cda8bc0da2f871ffea606ff71c80405725">
			<a href="https://teded.tumblr.com/post/184736320764/how-do-vaccines-work">https://teded.tumblr.com/post/184736320764/how-do-vaccines-work</a>
		</div>
		<script async src="https://assets.tumblr.com/post.js"></script>
	</div>
</figure>
```

```html
<figure class="wp-block-embed is-type-rich is-provider-tumblr wp-block-embed-tumblr">
	<div class="wp-block-embed__wrapper">
		<div class="tumblr-post" data-href="https://embed.tumblr.com/embed/post/2JT2XTaiTxO08wh21dqQrw/92003045635" data-did="7ce4825965cbd8bfd208f6aae43de7a528859aee">
			<a href="https://ifpaintingscouldtext.tumblr.com/post/92003045635/grant-wood-american-gothic-1930">https://ifpaintingscouldtext.tumblr.com/post/92003045635/grant-wood-american-gothic-1930</a>
		</div>
		<script async src="https://assets.tumblr.com/post.js"></script>
	</div>
</figure>
```

Reply:
Nevermind. It's for the reason I outlined in the issue 😄  The old regex didn't match the non-`\w` characters in the Tumblr URL.

Reply:
In any case, adding both as test cases will be good.